dt-core/admin/admin-settings-endpoints.php

```diff
@@ -222,6 +256,156 @@ public static function get_post_fields() {
+        $params = $request->get_params();
+        $updated = false;
+        if ( isset( $params['key'], $params['single'], $params['plural'], $params['displayed'] ) ){
+            $post_type = $params['key'];
+            $single = $params['single'];
+            $plural = $params['plural'];
+            $displayed = $params['displayed'];
+
+            // Fetch existing post type settings and update.
+            $custom_post_types = get_option( 'dt_custom_post_types', [] );
+
+            // Proceed with storing updates.
+            $post_type_settings = !empty( $custom_post_types[$post_type] ) ? $custom_post_types[$post_type] : [];
+            $post_type_settings['label_singular'] = $single;
+            $post_type_settings['label_plural'] = $plural;
+            $post_type_settings['hidden'] = ! $displayed;
+            $custom_post_types[$post_type] = $post_type_settings;
+
+            update_option( 'dt_custom_post_types', $custom_post_types );
+
+            $updated = true;
+        }
+
+        return [
+            'updated' => $updated
+        ];
+    }
+
+    public static function delete_post_type( WP_REST_Request $request ){
+        $params = $request->get_params();
+        $deleted = false;
+        $deleted_msg = '';
+        if ( isset( $params['key'] ) ){
+            $post_type = $params['key'];
+
+            // Ensure current user has the ability to delete post type;
+            if ( !current_user_can( 'delete_any_' . $post_type ) ){
+                return [
+                    'deleted' => false,
+                    'msg' => 'Current user unable to delete '. $post_type .' record type.'
+                ];
+            }
+
+            // Only process custom post types.
+            $custom_post_types = get_option( 'dt_custom_post_types', [] );
+            if ( isset( $custom_post_types[$post_type] ) ){
+
+                // Remove custom post type.
+                unset( $custom_post_types[$post_type] );
+                update_option( 'dt_custom_post_types', $custom_post_types );
+
+                // Remove field settings.
+                $field_customizations = dt_get_option( 'dt_field_customizations' );
+                if ( isset( $field_customizations[$post_type] ) ){
+                    unset( $field_customizations[$post_type] );
+
+                    update_option( 'dt_field_customizations', $field_customizations );
+                    wp_cache_delete( $post_type . '_field_settings' );
+                }
+
+                // If specified, proceed with deletion of all associated records.
+                $deleted_post_count = 0;
+                $deleted_post_counter = 0;
+                if ( isset( $params['delete_all_records'] ) && $params['delete_all_records'] ){
+                    global $wpdb;
+                    $posts = $wpdb->get_results( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = %s", $post_type ), ARRAY_A );
```

Comment:
@kodinkat please circle back to using the SQL we talked about or explain why we aren't using it.
This method will not work on a large number of posts.